dynawo/sources/Models/Modelica/Dynawo/Electrical/Transformers/BaseClasses_INIT.mo

```diff
@@ -160,4 +160,61 @@ equation
 
 end BaseTransformerVariableTap_INIT;
 
+partial model BaseGeneratorTransformer_INIT "Base model for initialization of GeneratorTransformer"
+
+/*
+  This model enables to initialize the generator model when the load-flow inputs are not known at the generator terminal but at the generator transformer terminal.
+
+  Equivalent circuit and conventions:
+
+               I1  r                I2
+    U1,P1,Q1 -->---oo----R+jX-------<-- U2,P2,Q2
+  (terminal1)                   |      (terminal2)
+                               G+jB
+                                |
+                               ---
+*/
+
+  import Dynawo.Electrical.SystemBase;
+
+  public
+
+    // Transformer parameters
+    parameter Types.ComplexImpedancePu ZPu = Complex(RPu, XPu) "Impedance in p.u (base U2Nom, SnRef)";
+    parameter Types.ComplexAdmittancePu YPu = Complex(GPu, BPu) "Admittance in p.u (base U2Nom, SnRef)";
+
+    // Start values at terminal (network terminal side)
+    parameter Types.ActivePowerPu P10Pu  "Start value of active power at terminal 1 in p.u (base SnRef) (receptor convention)";
+    parameter Types.ReactivePowerPu Q10Pu  "Start value of reactive power at terminal 1 in p.u (base SnRef) (receptor convention)";
+    parameter Types.VoltageModulePu U10Pu "Start value of voltage amplitude at terminal 1 in p.u (base U1Nom)";
+    parameter Types.Angle U1Phase0  "Start value of voltage angle at terminal 1 in rad";
+
+    Types.ComplexVoltagePu u10Pu "Start value of complex voltage at terminal 1 (base U1Nom)";
+    Types.ComplexApparentPowerPu s10Pu "Start value of complex apparent power at terminal 1 in p.u (base SnRef) (receptor convention)";
+    Types.ComplexCurrentPu i10Pu "Start value of complex current at terminal 1 (base U1Nom, SnRef) (receptor convention)";
+
+    Types.ComplexVoltagePu u20Pu "Start value of complex voltage at terminal 2 (base U2Nom)";
+    Types.ComplexCurrentPu i20Pu "Start value of complex current at terminal 2 (base U2Nom, SnRef) (receptor convention)";
+
+    Types.VoltageModulePu U20Pu "Start value of voltage amplitude at terminal 2 in p.u (base U2Nom)";
+    Types.ActivePowerPu P20Pu "Start value of active power at terminal 2 in p.u (base SnRef) (generator convention)";
+    Types.ReactivePowerPu Q20Pu "Start value of reactive power at terminal 2 in p.u (base SnRef) (generator convention)";
+    Types.Angle U2Phase0 "Start value of voltage angle in rad";
+
+  equation
+
+    s10Pu = Complex(P10Pu, Q10Pu);
+    u10Pu = ComplexMath.fromPolar(U10Pu, U1Phase0);
+    s10Pu = u10Pu * ComplexMath.conj(i10Pu);
+
+    rTfoPu * rTfoPu * u10Pu = rTfoPu * u20Pu + ZPu * i10Pu;
+    i10Pu = rTfoPu * (YPu * u20Pu - i20Pu);
```

Comment:
Again here the physical part (the two equations above) are the same in all the transformer initialization mode. Should we factorize them somewhere or keep them duplicated in different base models?

Reply:
factorization done